src/opcodes.js

```diff
@@ -0,0 +1,137 @@
+module.exports = {
+  // push value
+  OP_0: 0x00,
+  OP_FALSE: this.OP_0,
```

Comment:
`this.OP_0` is undefined.

Reply:
>     1. is the `opcodes.js` file even needed if we can get it from the `bscript-parser` project with the `opcodes.opcodeForWord` method?

Yeah I think that will work cleanly

>     What is the serialized script?  It appears to be an array of the bytes of the script.  Is that right?  An alternative to this would be to use the tokens from the bscript-parser directly.

It's a Buffer, but I'm open to alerternatives. The tokens are strings, correct?

>     3\. The switch statement for all the opcodes works well enough, but an alternative might be a map (an object) with opcodes as the key and an execution function as the value

Yeah that sounds like a better idea

Reply:
> It's a Buffer, but I'm open to alerternatives. The tokens are strings, correct?

The tokens are Token objects: https://github.com/JBaczuk/bscript-parser/blob/master/src/token.js

Reply:
Ok, and what do the types mean?

```
Token.LITERAL = 'LITERAL'
Token.OPCODE = 'OPCODE'
Token.PLACEHOLDER = 'PLACEHOLDER'
```

Reply:
A `Token.LITERAL` token is any data being pushed onto the stack with  `PUSH_DATA(n)` opcode.  The `value` property will be a buffer of the pushed data in this case. 

A `Token.OPCODE` token is for any non-`PUSH_DATA(n)` opcode and the `value` property will be the integer value of the opcode.

A `Token.PLACEHOLDER` token is only used for template scripts.  Template scripts are used internally to detect script types.

Reply:
I think I understand your vision better now.